File: backend/app/schemas/insights.py
from pydantic import BaseModel


class SignalsProcessed(BaseModel):
    today: int = 0
    this_week: int = 0
    all_time: int = 0


class IntelligenceTip(BaseModel):
    title: str
    description: str
    severity: str = "info"  # info, warning, critical


class TrendingTopic(BaseModel):
    topic: str
    count: int


class ActivitySpike(BaseModel):
    channel_id: str
    channel_title: str
    today_count: int
    daily_average: float
    ratio: float


class InsightsDashboardResponse(BaseModel):
    signals_processed: SignalsProcessed
    intelligence_tips: list[IntelligenceTip]
    trending_topics: list[TrendingTopic]
    activity_spikes: list[ActivitySpike]

File: frontend/src/lib/api/insights.ts
import { api } from './axios-instance'

export interface SignalsProcessed {
  today: number
  this_week: number
  all_time: number
}

export interface IntelligenceTip {
  title: string
  description: string
  severity: 'info' | 'warning' | 'critical'
}

export interface TrendingTopic {
  topic: string
  count: number
}

export interface ActivitySpike {
  channel_id: string
  channel_title: string
  today_count: number
  daily_average: number
  ratio: number
}

export interface InsightsDashboardData {
  signals_processed: SignalsProcessed
  intelligence_tips: IntelligenceTip[]
  trending_topics: TrendingTopic[]
  activity_spikes: ActivitySpike[]
}

export const insightsApi = {
  dashboard: () => api.get<InsightsDashboardData>('/api/insights/dashboard'),
}
